packages/dotenv-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,92 @@
+const fs = require("fs");
+const dotenv = require("dotenv");
+const dotenvExpand = require("dotenv-expand");
+const { DefinePlugin } = require("webpack");
+
+/** @typedef {import("./types").Config} Config */
+
+class DotenvWebpackPlugin {
+  /**
+   * Dotenv Webpack Plugin
+   * @param {Config} config - Configuration options
+   */
+  constructor(config = {}) {
+    const currentDirectory = process.cwd();
+
+    this.options = Object.assign(
+      {},
+      {
+        envFiles: [
+          `${currentDirectory}/.env.example`, // loaded in all cases
+          `${currentDirectory}/.env`, // loaded in all cases
+          `${currentDirectory}/.env.local`, // loaded in all cases, ignored by git
+          `${currentDirectory}/.env.[mode]`, // only loaded in specified mode
+          `${currentDirectory}/.env.[mode].local`, // only loaded in specified mode, ignored by git
+        ],
+        prefixes: ["process.env.", "import.meta.env."],
+      },
+      config,
+    );
+  }
+
+  /**
+   * Webpack apply hook
+   * @param {Object} compiler - Webpack compiler
+   * @returns {void}
+   */
+  apply(compiler) {
+    const mode = compiler.options.mode || "production";
+
+    // throw error if envFiles is not an array
+    if (!Array.isArray(this.options.envFiles)) {
+      const logger = compiler.getInfrastructureLogger("DotenvWebpackPlugin");
+      logger.error(`envFiles option must be an array, received ${typeof this.options.envFiles}`);
+      return;
+    }
```

Comment:
Avoid this and please use https://github.com/webpack-contrib/terser-webpack-plugin/blob/master/src/index.js#L193, same for any validation

Reply:
done. now using schema utils for validation